functional tests not working because of a WORM protected file in play.min.io

## Expected behavior

Run all functional tests without errors with default parameters

## Actual behavior

During a functional tests I get:

```
Running functional tests
Dependency validation complete
Running test_make_bucket()
Running test_make_bucket_error()
Running test_rb()
mc: test_rb: /root/mc/mc --config-dir /tmp/.mc-21807 --quiet --no-color rb --force --dangerous myminio  >>> 
mc: <ERROR> Failed to remove `myminio/12345-flwo-whatever`. Object, 'Čo tebe do toho (Version ID=eca82f19-4353-4970-9fbe-a6ea4a1a0c6c)' is WORM protected and cannot be overwritten.
make: *** [Makefile:46: test] Error 1
```

## Steps to reproduce the behavior

`make test`

If I comment this line in `functional_tests.sh`, everything runs fine:

```
assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd rb --force --dangerous "${SERVER_ALIAS}"
```

I get this error because in the `play.min.io` instance, used by default, someone put a WORM protected file named `Čo tebe do toho` in the `12345-flwo-whatever` bucket. So we can't delete all the content of the site. Maybe this test should be disabled in such default case.

## mc --version

mc version DEVELOPMENT.2024-03-13T23-51-57Z (commit-id=2508db9c560c10be0ed2203eaa585134235b6907)
Runtime: go1.22.1 linux/arm64
Copyright (c) 2015-2024 MinIO, Inc.
License GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>

## System information

Multiple linux distributions (amd64 or arm64)

--- a/functional-tests.sh
+++ b/functional-tests.sh
@@ -65,7 +65,7 @@ if [ -z "${SERVER_ENDPOINT+x}" ]; then
 fi
 
 # If you want to run the complete site cleaning test, set this variable to true 
-COMPLETE_RB_TEST=false
+COMPLETE_RB_TEST=true
 
 WORK_DIR="$PWD"
 DATA_DIR="$MINT_DATA_DIR"

--- a/functional-tests.sh
+++ b/functional-tests.sh
@@ -278,7 +278,7 @@ function test_rb()
     assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd rb --force "${SERVER_ALIAS}/${bucket1}"
 
     # Test rb with --force and --dangerous to remove a site content
-    if [ "${COMPLETE_RB_TEST}" = "true" ]; then
+    if [ "${COMPLETE_RB_TEST}" == "true" ]; then
         assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd mb "${SERVER_ALIAS}/${bucket1}"
         assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd mb "${SERVER_ALIAS}/${bucket2}"
         assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd cp "${FILE_1_MB}" "${SERVER_ALIAS}/${bucket1}/${object_name}"

--- a/functional-tests.sh
+++ b/functional-tests.sh
@@ -64,6 +64,9 @@ if [ -z "${SERVER_ENDPOINT+x}" ]; then
     ENABLE_HTTPS=1
 fi
 
+# If you want to run the complete site cleaning test, set this variable to true 
+COMPLETE_RB_TEST=false
+
 WORK_DIR="$PWD"
 DATA_DIR="$MINT_DATA_DIR"
 if [ -z "$MINT_MODE" ]; then
@@ -275,13 +278,15 @@ function test_rb()
     assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd rb --force "${SERVER_ALIAS}/${bucket1}"
 
     # Test rb with --force and --dangerous to remove a site content
-    assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd mb "${SERVER_ALIAS}/${bucket1}"
-    assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd mb "${SERVER_ALIAS}/${bucket2}"
-    assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd cp "${FILE_1_MB}" "${SERVER_ALIAS}/${bucket1}/${object_name}"
-    assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd cp "${FILE_1_MB}" "${SERVER_ALIAS}/${bucket2}/${object_name}"
-    assert_failure "$start_time" "${FUNCNAME[0]}" mc_cmd rb --force "${SERVER_ALIAS}/"
-    assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd rb --force --dangerous "${SERVER_ALIAS}"
-
+    if [ "${COMPLETE_RB_TEST}" = "true" ]; then
+        assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd mb "${SERVER_ALIAS}/${bucket1}"
+        assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd mb "${SERVER_ALIAS}/${bucket2}"
+        assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd cp "${FILE_1_MB}" "${SERVER_ALIAS}/${bucket1}/${object_name}"
+        assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd cp "${FILE_1_MB}" "${SERVER_ALIAS}/${bucket2}/${object_name}"
+        assert_failure "$start_time" "${FUNCNAME[0]}" mc_cmd rb --force "${SERVER_ALIAS}/"
+        assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd rb --force --dangerous "${SERVER_ALIAS}"
+    fi
+    
     log_success "$start_time" "${FUNCNAME[0]}"
 }
 

